src/main/java/com/simibubi/create/content/processing/basin/BasinRecipe.java

```diff
@@ -147,8 +149,27 @@ private static boolean apply(BasinBlockEntity basin, Recipe<?> recipe, boolean t
 			}
 
 			if (simulate) {
-				CraftingInput remainderInput = new DummyCraftingContainer(availableItems, extractedItemsFromSlot)
-					.asCraftInput();
+				CraftingInput remainderInput = CraftingInput.of(3, 3,
+					IntStream.range(0, availableItems.getSlots())
```

Comment:
`apply` is a hotspot method as it's also used to match the recipe, maybe switch to normal `fori` loops to reduce the overhead.